src/server/scripts/Northrend/AzjolNerub/ahnkahet/boss_elder_nadox.cpp

```diff
@@ -202,23 +203,89 @@ class boss_elder_nadox : public CreatureScript
                 }
                 case EVENT_CHECK_HOME:
                 {
-                    if (me->HasAura(SPELL_ENRAGE))
-                        break;
-
-                    if (me->GetPositionZ() < 24)
+                    if (!me->HasAura(SPELL_ENRAGE) && (me->GetPositionZ() < 24.0f || !me->GetHomePosition().IsInDist(me->GetPositionX(), me->GetPositionY(), me->GetPositionZ(), 110.0f)))
                     {
-                        me->CastSpell(me, SPELL_ENRAGE, true);
+                        DoCastSelf(SPELL_ENRAGE, true);
                         events.PopEvent();
                         break;
                     }
 
                     events.RepeatEvent(2000);
-                    break;
+                }break;
                 }
```

Comment:
I think you know the drill by now :D

Reply:
will rework it today, this entire script needs rewrite (still)